### PR TITLE
The default android platform settings refer to lumberyard #3881

### PR DESCRIPTION
**Pull request by jjjoness**
Fixes bug #3881

### CODE REVIEW

File: cmake/Tools/Platform/Android/unit_test_android_deployment.py
Comment (AMZN-ScottR) [inline cmake/Tools/Platform/Android/unit_test_android_deployment.py:596]: Could remove the ".amazon" portion as well while you're here

File: cmake/Tools/Platform/Android/unit_test_android_deployment.py
Comment (AMZN-ScottR) [inline cmake/Tools/Platform/Android/unit_test_android_deployment.py:668]: Same here regarding the ".amazon" portion

File: cmake/Tools/Platform/Android/unit_test_android_deployment.py
Comment (AMZN-ScottR) [inline cmake/Tools/Platform/Android/unit_test_android_deployment.py:764]: Here too

File: Code/Framework/AzAndroid/java/com/amazon/lumberyard/LumberyardActivity.java
Comment (spham-amzn) [inline Code/Framework/AzAndroid/java/com/amazon/lumberyard/LumberyardActivity.java:67]: We should replace all `com.o3de.samples` with `org.o3de.samples` to reflect `03de.org`, (https://www.oracle.com/java/technologies/javase/codeconventions-namingconventions.html )

File: cmake/Tools/Platform/Android/unit_test_android_deployment.py
Comment (spham-amzn) [inline cmake/Tools/Platform/Android/unit_test_android_deployment.py:596]: These still all have 'amazon'. `com.amazon.o3de.*` can be replaced with `org.o3de` instead

File: cmake/Tools/Platform/Android/unit_test_android_deployment.py
Comment (spham-amzn) [inline cmake/Tools/Platform/Android/unit_test_android_deployment.py:668]: same comment as above

File: cmake/Tools/Platform/Android/unit_test_android_deployment.py
Comment (spham-amzn) [inline cmake/Tools/Platform/Android/unit_test_android_deployment.py:764]: Same comment as above

File: cmake/Tools/common.py
Comment (spham-amzn) [inline cmake/Tools/common.py:125]: nit on grammar
```suggestion
    Read an o3de config file and extract specific keys if they are set
```

File: cmake/Tools/unit_test_common.py
Comment (spham-amzn) [inline cmake/Tools/unit_test_common.py:176]:
```suggestion
        "package_name" : "org.o3de.yourgame",
```

**Review by AMZN-ScottR — COMMENTED** [review]
Thanks for doing this!

Couple other things you could do while here:
- updating the "com." prefix to "org." instead
- updating CFBundleIdentifier for the Apple plist files to match

File: Templates/DefaultProject/Template/Platform/Android/android_project.json
Comment (lumberyard-employee-dm) [inline Templates/DefaultProject/Template/Platform/Android/android_project.json:4]: Is it really correct to use the "o3de" naming scheme in the Project Templates?
Seems like this should probably not be set

Comment (AMZN-ScottR) [reply Templates/DefaultProject/Template/Platform/Android/android_project.json:4]: The Android Gradle project generation unfortunately requires this here

Comment (spham-amzn) [reply Templates/DefaultProject/Template/Platform/Android/android_project.json:4]: yeah, maybe that could be an additional argument to the android project generator and that this file is used as a template instead

Comment (AMZN-ScottR) [reply Templates/DefaultProject/Template/Platform/Android/android_project.json:4]: That would certainly work, may be on the edge of this change's scope though.

In an ideal world, the Android project builder files would be included in the template instead and things like this would go away but that requires even more and significant changes.

File: Code/Framework/AzAndroid/java/com/amazon/lumberyard/LumberyardActivity.java
Comment (jjjoness) [reply Code/Framework/AzAndroid/java/com/amazon/lumberyard/LumberyardActivity.java:67]: Done.

File: cmake/Tools/Platform/Android/unit_test_android_deployment.py
Comment (jjjoness) [reply cmake/Tools/Platform/Android/unit_test_android_deployment.py:596]: Done.

File: cmake/Tools/Platform/Android/unit_test_android_deployment.py
Comment (jjjoness) [reply cmake/Tools/Platform/Android/unit_test_android_deployment.py:668]: Done.

File: cmake/Tools/common.py
Comment (jjjoness) [reply cmake/Tools/common.py:125]: Done.

File: cmake/Tools/unit_test_common.py
Comment (jjjoness) [reply cmake/Tools/unit_test_common.py:176]: Done.

File: cmake/Tools/Platform/Android/unit_test_android_deployment.py
Comment (jjjoness) [reply cmake/Tools/Platform/Android/unit_test_android_deployment.py:596]: Done.

File: cmake/Tools/Platform/Android/unit_test_android_deployment.py
Comment (jjjoness) [reply cmake/Tools/Platform/Android/unit_test_android_deployment.py:668]: Done.

File: Templates/DefaultProject/Template/Platform/Android/android_project.json
Comment (jjjoness) [reply Templates/DefaultProject/Template/Platform/Android/android_project.json:4]: Who should decide what this should be?

**Review by spham-amzn — APPROVED** [review]

File: Templates/DefaultProject/Template/Platform/Android/android_project.json
Comment (AMZN-ScottR) [reply Templates/DefaultProject/Template/Platform/Android/android_project.json]: I'd say the the change to "org.o3de.${Name}" is fine for now, anything beyond that will require some careful consideration.

Comment (lumberyard-employee-dm) [reply Templates/DefaultProject/Template/Platform/Android/android_project.json]: I am find with it being `"org.o3de.${Name}"` for now

**Review by AMZN-ScottR — APPROVED** [review]
Would be great to see these bundle/package updates to the following as well:
- Templates/DefaultProject/Template/Resources/Platform/iOS/Info.plist
- Templates/DefaultProject/Template/Resources/Platform/Mac/Info.plist
- Templates/MinimalProject/Template/Resources/Platform/iOS/Info.plist
- Templates/MinimalProject/Template/Resources/Platform/Mac/Info.plist

Won't hold up what you have here though.  Thanks again for this!

**Review by lumberyard-employee-dm — APPROVED** [review]

File: Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/ActionBus.h
Comment (AMZN-alexpete) [inline Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/ActionBus.h:30]: should we switch these to `AZ_CRC_CE`, or re-run the precompile step to generate the updated CRC32s?

File: Gems/PhysX/Code/Editor/ColliderComponentMode.cpp
Comment (AMZN-alexpete) [inline Gems/PhysX/Code/Editor/ColliderComponentMode.cpp:33]: Ditto

File: Gems/LmbrCentral/Code/Source/Shape/EditorTubeShapeComponentMode.cpp
Comment (AMZN-alexpete) [inline Gems/LmbrCentral/Code/Source/Shape/EditorTubeShapeComponentMode.cpp:25]: Ditto

File: Gems/WhiteBox/Code/Source/SubComponentModes/EditorWhiteBoxDefaultMode.cpp
Comment (AMZN-alexpete) [inline Gems/WhiteBox/Code/Source/SubComponentModes/EditorWhiteBoxDefaultMode.cpp:37]: Ditto

**Review by AMZN-alexpete — APPROVED** [review]
Nice! Just a minor comment about whether we should switch the AZ_CRCs to AZ_CRC_CE.

File: Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/ActionBus.h
Comment (lumberyard-employee-dm) [reply Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/ActionBus.h:30]: There is no precompile step to geneate CRC32 values anymore as that would modify the source file and invalidate the build.
AZ_CRC_CE can be used

**Pull request merged**
